akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/ibex35/Ibex35StatManager.scala

```diff
@@ -0,0 +1,264 @@
+import com.tapiporla.microservices.retrievers.indices.ibex35.model.{Ibex35Historic, Ibex35Stat}
+import org.elasticsearch.search.sort.SortOrder
+import org.joda.time.DateTime
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+
+object Ibex35StatManager {
+  object InitIbex35StatManager //Called after Ibex35HistoricManager is updated to avoid race conditions
+  object UpdateStats //Called to Update stats (after Ibex35HistoricManager has inserted new data)
+  object CheckReadyToStart //Called to check if we have all the data needed to Start the manager
+  object StatsUpdatedSuccessfully //When stats has been generated succesfully
+}
+
+/**
+  * In charge of create stats on demand from the index
+  *
+  * The flow is: initial -> gettingInitialStatus -> waitingForDeletions -> ready -> updating -> ready...
+  *
+  */
+class Ibex35StatManager extends Actor with ActorLogging with Stash {
+
+  val esDAO =
+    context.actorOf(Props[Ibex35ESDAO], name = "Ibex35ESDAO_Manager")
+
+  override def receive: Receive = initial
+
+  def initial: Receive = {
+
+    case InitIbex35StatManager =>
+      log.info("Recovering initial status")
+
+      //Recover MM*** status
+      esDAO ! retrieveLastMMUpdated(StatsGenerator.MM200)
+      esDAO ! retrieveLastMMUpdated(StatsGenerator.MM100)
+      esDAO ! retrieveLastMMUpdated(StatsGenerator.MM40)
+      esDAO ! retrieveLastMMUpdated(StatsGenerator.MM20)
+      context.become(gettingInitialStatus((false, None), (false, None), (false, None), (false, None)))
+
+    case _ =>
+      stash()
+  }
+
+  type CheckedMM = (Boolean, Option[DateTime])
+
+  def gettingInitialStatus(
+                          lastMM200: CheckedMM,
+                          lastMM100: CheckedMM,
+                          lastMM40: CheckedMM,
+                          lastMM20: CheckedMM
+                          ): Receive = {
+
+    case DataRetrieved(_, _, data, rq) if rq.where.nonEmpty =>
+      rq.where.get match {
+        case TermQueryDefinition(Ibex35ESDAO.Stats.statsAttr, StatsGenerator.MM200._1, _, _) =>
+            context.become(
+                gettingInitialStatus(generateCheckedMM(data), lastMM100, lastMM40, lastMM20)
+              )
+
+        case TermQueryDefinition(Ibex35ESDAO.Stats.statsAttr, StatsGenerator.MM100._1, _, _) =>
+          context.become(
+            gettingInitialStatus(lastMM200, generateCheckedMM(data), lastMM40, lastMM20)
+          )
+
+        case TermQueryDefinition(Ibex35ESDAO.Stats.statsAttr, StatsGenerator.MM40._1, _, _) =>
+          context.become(
+            gettingInitialStatus(lastMM200, lastMM100, generateCheckedMM(data), lastMM20)
+          )
+
+        case TermQueryDefinition(Ibex35ESDAO.Stats.statsAttr, StatsGenerator.MM20._1, _, _) =>
+          context.become(
+            gettingInitialStatus(lastMM200, lastMM100, lastMM40, generateCheckedMM(data))
+          )
+
+        case _ => log.error("Unknown where to threat!!")
```

Comment:
If the query is not expected, should it also send to self the ```CheckReadyToStart``` message?

Reply:
Removed with unhandled now